optimum/habana/transformers/models/gptj/modeling_gptj.py

```diff
@@ -120,10 +220,29 @@ def forward(
                 key = torch.cat([past_key, key], dim=-2)
                 value = torch.cat([past_value, value], dim=-2)
 
-        if use_cache is True:
+        if use_cache is True and token_idx is not None:
+            if reuse_cache:
+                key = self.k_cache(key, 2, token_idx)
+                value = self.v_cache(value, 2, token_idx)
+                present = (self.k_cache.get_shape(), self.v_cache.get_shape())
+            else:
+                if layer_past is None:
+                    past_key = torch.zeros(key.shape, dtype=self.k_proj.weight.dtype, device=key.device)
+                    past_value = torch.zeros(key.shape, dtype=self.k_proj.weight.dtype, device=key.device)
+                    layer_past = (past_key, past_value)
+                key = self.k_cache.update(layer_past[0], key, 2, token_idx, self.inp_seq_len)
+                value = self.v_cache.update(layer_past[1], value, 2, token_idx, self.inp_seq_len)
+                present = layer_past
+
+            if cache_idx is not None and q_len == 1:
```

Comment:
cache_idx will help with long output such as 2048 case.
You need to add --bucket_internal --bucket_size 128 for 2024 output test

Reply:
Ok, thanks. I will add to 2048 output test cases.